[ADD] crm_event: combine leads and event types

yajo opened the pull request:
This module extends the functionality of CRM opportunities (and leads, if
enabled) to support linking them to event types and to allow you to keep track
of leads interested in an upcoming event of some type.

This is useful if you organize your events based on the amount of people
interested in a certain type of event.

@Tecnativa TT27664

# Pictures

Just to showcase the features, so you know what to test on runbot.

Let you specify event type and wanted seats in opportunities:

![imagen](https://user-images.githubusercontent.com/973709/105829646-8da7b200-5fbc-11eb-9ed6-2bc76ffe513f.png)

Quick event type search on opportunities:

![imagen](https://user-images.githubusercontent.com/973709/105829541-681aa880-5fbc-11eb-830a-2e58e1fd39ed.png)

Quick group by event type:

![imagen](https://user-images.githubusercontent.com/973709/105829596-7a94e200-5fbc-11eb-8496-9d8f17bacdf2.png)

On pipeline reporting, you can filter by event-related opportunities, group by event type, and measure by wanted seats:

![imagen](https://user-images.githubusercontent.com/973709/105829891-cc3d6c80-5fbc-11eb-8a2c-bc60cfdc6b5c.png)

The same of the above, but for leads (if they are enabled in the settings).


Smart button in event types form, indicating open events of that type (and seats available) and open opportunities of that type (and wanted seats):

![imagen](https://user-images.githubusercontent.com/973709/105829334-27bb2a80-5fbc-11eb-940f-7d325c5d0e18.png)

pedrobaeza commented:
Call the module `crm_event` for not limiting to event types. The integration can go further in any time.

yajo commented:
Renamed.

sergio-teruel commented on crm_event/models/crm_lead.py:
```diff
@@ -0,0 +1,26 @@
+            "If this lead/opportunity is related to a specific event type, "
+            "indicate it here."
+        ),
+    )
```
Maybe woulb be good idea add an ondelete??

yajo replied:
[It's already there](https://github.com/OCA/event/pull/201/files/816ff9125b7c7018c4069ba735912302b97f71a9#diff-6cdd1fe21981658519522db80332ba2dd1899d579ce9b167c5267da1e214057bR13).

sergio-teruel reviewed (approved):
Only little comment

chienandalu reviewed (approved):
:+1:

pedrobaeza commented on crm_event/views/event_type_view.xml:
```diff
@@ -0,0 +1,30 @@
+                    event_sale_reservation -->
+                <div class="oe_button_box oe_inline" name="button_box">
+                    <!-- Use same iconography as in contact form -->
+                    <button colspan="2" class="oe_stat_button" type="action" name="%(event.action_event_view)d" icon="fa-ticket" context="{'default_event_type_id': id, 'search_default_event_type_id': id}">
```
Can you switch this to a method for avoiding the problem with `id`?

yajo replied:
Done.

yajo commented:
Well, this is approved an all comments attended, so...

/ocabot merge nobump

yajo commented:
Sorry, I just had to add a few features before the merge:

![imagen](https://user-images.githubusercontent.com/973709/107509136-f5075980-6b99-11eb-97ab-97163f6f8967.png)

![imagen](https://user-images.githubusercontent.com/973709/107509326-4d3e5b80-6b9a-11eb-9cc5-deedaf8a76e8.png)

Commit:
[ADD] crm_event: combine leads and event types

This module extends the functionality of CRM opportunities (and leads, if
enabled) to support linking them to event types and to allow you to keep track
of leads interested in an upcoming event of some type.

This is useful if you organize your events based on the amount of people
interested in a certain type of event.

@Tecnativa TT27664

yajo commented:
New and hopefully last features:

![imagen](https://user-images.githubusercontent.com/973709/107750483-007e9000-6d14-11eb-99df-576d3e7f93a6.png)
![imagen](https://user-images.githubusercontent.com/973709/107750521-0d9b7f00-6d14-11eb-9ea1-576ed40fd0c3.png)

pedrobaeza commented:
/ocabot merge nobump
